src/__tests__/defaultRenderer.pure.test.ts

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { ReactHooksRenderer } from '../types/react'
+
+describe('default renderer', () => {
+  beforeEach(() => {
+    jest.resetModules()
+  })
+
+  test('should resolve native renderer as default renderer', () => {
+    const expectedRenderer = require('../native/pure') as ReactHooksRenderer
+    const actualRenderer = require('../pure') as ReactHooksRenderer
```

Comment:
this was the only test where I felt the `require`s in the tests were still appropriate.  I worked around the issue by changing to use the `pure` imports instead.   This left the non-`pure` import uncovered, which is why I introduced the [simplified test file](https://github.com/testing-library/react-hooks-testing-library/pull/639/files#diff-6ad1e6bacfc4e8e27309955adc84b9596acd1c31bdbb575dd8cefbf6ced6a454) for it.